### ISSUE TITLE
fitView() doesn't work as expected in onInit() callback

### ISSUE THREAD

**Issue by bsvetlik** [xyflow/xyflow#4793]
### What platform were you using when you found the bug?

- React Flow / Svelte Flow version: @xyflow-12.3.4
- Browser and version: Latest Chrome, Edge, FireFox
- OS and version: Windows 10

### Describe the Bug

We want to automatically arrange nodes in a hierarchical layout and fit them to view after initialization:

```
const onInit = () => {
    const { nodes: layoutedNodes, edges: layoutedEdges } = layoutGraph(nodes, edges); // dagre auto layout
    setNodes([...layoutedNodes]);
    setEdges([...layoutedEdges]);

    fitView( { maxZoom: 1, minZoom: 0.1, duration: 0 }); // has no effect
};
```

The fitView() function is not functioning as expected within the onInit() callback. To compensate for this, we've implemented a temporary solution involving a 100-millisecond delay using setTimeout. Unfortunately, this approach is unreliable not sufficient in all cases.

### Steps to reproduce the bug or issue

See above code.
This issue may be related to #4792

### Expected behavior

Ensure that the fitView() function works reliably in all scenarios, including initial rendering.

**Comment by bsvetlik** [comment]
@peterkogo 
I apologize for the incorrect code I provided earlier to illustrate the fitView() issue in React Flow.  While I appreciate your example, it doesn't quite address the specific problem I'm encountering.

I've made some adjustments to the code and uploaded a revised version to CodeSandbox: https://codesandbox.io/p/sandbox/condescending-gagarin-vdkxqp
Changed line: `position: { x: index * node.position.x, y: index * node.position.y },`

If you could refresh the browser a few times, you'll see that the fitView() function doesn't always work as expected.

I'll try to create a more specific CodeSandbox example tomorrow that better demonstrates the issue we're facing with fitView().

**Comment by bsvetlik** [comment]
@peterkogo
I've made some modifications to your example code to ensure fitView() works consistently after each browser refresh. You can find the corrected code here: https://codesandbox.io/p/sandbox/upbeat-proskuriakova-ydty3l

The similar solution is implemented in our application, but unfortunately, fitView() exhibits inconsistent behavior, just like in your initial example.

While I haven't been able to replicate the exact problem within CodeSandbox, I suspect it might be connected to the size of the flow container not being defined during initialization.  I'll continue investigating this possibility.

**Comment by moklick** [comment]
This is fixed in 12.5.0. You no longer need to wrap `fitView` with `setTimeout` or `requestAnimtionFrame`.

**Comment by bsvetlik** [comment]
I upgraded to version 12.6.4, and it works as expected now. Great job!
